src/manifest/adaptation.ts

```diff
@@ -143,6 +145,10 @@ export default class Adaptation {
       this.isDub = parsedAdaptation.isDub;
     }
 
+    this.trickModeTracks = trickModeTracks !== undefined ?
+      trickModeTracks.map((track) => new Adaptation(track)) :
+      undefined;
```

Comment:
By reading it, I'm hesitating between what you wrote and:
```js
if (trickModeTracks !== undefined && trickModeTracks.length > 0) {
  this.trickModeTracks = trickModeTracks.map((track) => new Adaptation(track));
}
```
Here we would only define the variable if it is defined, which might be better for people debugging in the console.
This is a very tiny nitpick though, as I like also how you wrote it. You're the judge.